### PR TITLE
[Reviews] nullable title for reviews

### LINKED ISSUE #8172: [Review] Why title is required ?
On many websites, you can rate something without comment it.
On Review model, comment is nullable but title is not, it's weird. On few projects, I use review bundle in this use case and I needed to work around.

What do you think ?

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes #8172 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ReviewBundle/Resources/config/validation/Review.xml [hunk 1]
```diff
@@ -17,7 +17,7 @@
             <constraint name="NotBlank">
```
Comment (pamil) [inline src/Sylius/Bundle/ReviewBundle/Resources/config/validation/Review.xml]: Shouldn't this constraint be removed now?

Comment (loic425) [reply src/Sylius/Bundle/ReviewBundle/Resources/config/validation/Review.xml]: I don't think so. IMHO, when you write a commented review, you need a title and a comment, when you have a "rating system" by only clicking on stars for example, title is not required.

Comment (pamil) [reply src/Sylius/Bundle/ReviewBundle/Resources/config/validation/Review.xml]: Does it work like that now (title is required if comment is filled)? I guess we'd need some Behat scenario to cover that.

Comment (loic425) [reply src/Sylius/Bundle/ReviewBundle/Resources/config/validation/Review.xml]: No, I just kept existing behaviour on sylius core :)
We can go further but it's more complex. For exemple, you need to separate ratings from commented reviews on frontend. Moreover, the status workflow is different. A single rate doesn't need to be accepted.

Comment (loic425) [reply src/Sylius/Bundle/ReviewBundle/Resources/config/validation/Review.xml]: @pamil I think it's just a first step for sylius. It's just an improvment for sylius review bundle and an easier way to improve sylius review behaviour in a sylius shop.

Comment (loic425) [reply src/Sylius/Bundle/ReviewBundle/Resources/config/validation/Review.xml]: @Zales0123 @pamil @pjedrzejewski @lchrusciel 
Could you think about that, discuss this and tell me what do you think about the best behaviour I could implement.

Currently, with existing Pull request, I added a new validation group (sylius_review) on title to allow using only "sylius" validation group. Another solution could be the idea suggested by Pamil: title is required only if comment is filled. I guess it needs a form listener to do that.

Comment (loic425) [reply src/Sylius/Bundle/ReviewBundle/Resources/config/validation/Review.xml]: Any idea?

Comment (Zales0123) [reply src/Sylius/Bundle/ReviewBundle/Resources/config/validation/Review.xml]: Sorry @loic425 for the long response. I thing that requiring comment only if title is filled is the best approach, just seems as the most natural solution. /cc @pamil @pjedrzejewski @lchrusciel

Comment (loic425) [reply src/Sylius/Bundle/ReviewBundle/Resources/config/validation/Review.xml]: @Zales0123 Ok, I'm gonna work on that approach.

**Comment by pamil** [comment]
Thanks Loic!

**Comment by pamil** [comment]
I've created #9239 to track that conditional validation feature.

**Pull request merged**